be/test/exprs/map_functions_test.cpp

```diff
@@ -319,4 +328,129 @@ PARALLEL_TEST(MapFunctionsTest, test_map_function) {
+    type_map_int_int.children.emplace_back(TypeDescriptor(LogicalType::TYPE_INT));
+
+    auto map_column_nullable = ColumnHelper::create_column(type_map_int_int, true);
+    {
+        //   [1->44, 2->55, 4->66]
+        //   [2->77, 3->88]
+        //   [3 -> NULL]
+        //   []
+        //   NULL
+        DatumMap map1;
+        map1[(int32_t)1] = (int32_t)44;
+        map1[(int32_t)2] = (int32_t)55;
+        map1[(int32_t)4] = (int32_t)66;
+        map_column_nullable->append_datum(map1);
+
+        DatumMap map2;
+        map2[(int32_t)2] = (int32_t)77;
+        map2[(int32_t)3] = (int32_t)88;
+        map_column_nullable->append_datum(map2);
+
+        DatumMap map3;
+        map3[(int32_t)3] = Datum();
+        map_column_nullable->append_datum(map3);
+
+        // {} empty
+        map_column_nullable->append_datum(DatumMap());
+        // NULL
+        map_column_nullable->append_datum(Datum{});
+    }
+
+    auto map_column_not_nullable = ColumnHelper::create_column(type_map_int_int, false);
+    {
+        //   [1->44, 2->55, 4->66]
+        //   [2->77, 3->88]
+        //   [3 -> NULL]
+        //   []
+        //   []
+        DatumMap map1;
+        map1[(int32_t)1] = (int32_t)44;
+        map1[(int32_t)2] = (int32_t)55;
+        map1[(int32_t)4] = (int32_t)66;
+        map_column_not_nullable->append_datum(map1);
+
+        DatumMap map2;
+        map2[(int32_t)2] = (int32_t)77;
+        map2[(int32_t)3] = (int32_t)88;
+        map_column_not_nullable->append_datum(map2);
+
+        DatumMap map3;
+        map3[(int32_t)3] = Datum();
+        map_column_not_nullable->append_datum(map3);
+
+        // {} empty
+        map_column_not_nullable->append_datum(DatumMap());
+        map_column_not_nullable->append_datum(DatumMap());
+    }
+
+    TypeDescriptor TYPE_ARRAY_BOOLEAN = create_array_type(TYPE_BOOLEAN);
+
+    // [null, true, false]
+    // []
+    // [false]
+    // [NULL]
+    // [true]
+    auto bool_array_not_nullable = ColumnHelper::create_column(TYPE_ARRAY_BOOLEAN, false);
+    bool_array_not_nullable->append_datum(DatumArray{Datum(), true, false});
+    bool_array_not_nullable->append_datum(DatumArray{});
+    bool_array_not_nullable->append_datum(DatumArray{false});
+    bool_array_not_nullable->append_datum(DatumArray{Datum()});
+    bool_array_not_nullable->append_datum(DatumArray{true});
+
+    // [null, true, false]
+    // NULL
+    // [false]
+    // [null]
+    // [false, null]
+    auto bool_array_nullable = ColumnHelper::create_column(TYPE_ARRAY_BOOLEAN, true);
+    bool_array_nullable->append_datum(DatumArray{Datum(), true, false});
+    bool_array_nullable->append_datum(Datum{});
+    bool_array_nullable->append_datum(DatumArray{false});
+    bool_array_nullable->append_datum(DatumArray{Datum()});
+    bool_array_nullable->append_datum(DatumArray{false, Datum()});
+
+    {
+        auto result = MapFunctions::map_filter(nullptr, {map_column_nullable, bool_array_not_nullable}).value();
```

Comment:
How about write these test cases with the gtest like the `json_functions_test.cpp` ?

Reply:
here already cover error prone cases like array_filter.